app/screens/interactive_dialog/dialog_element.js

```diff
@@ -158,7 +158,7 @@ export default class DialogElement extends PureComponent {
                 <BoolSetting
                     id={name}
                     label={displayName}
-                    value={value === 'true'}
+                    value={Boolean(value)}
```

Comment:
Toggling works now but there's a discrepancy on initial render when the default value (which is of [type string](https://docs.mattermost.com/developer/interactive-dialogs.html#checkbox-element)) is `"False"` or `"false"`.
Please take a look again.

Reply:
Ah, that's a truly unfortunate requirement from the data model. I'll extend the demo plugin with default cases here, tweak, and test again.

Reply:
It looks like the webapp is broken here too in the same way. That, and we show the "required bit" for these fields, but it's not really clear what that means (required `true`?). I'll work towards patching all of it.